### PR TITLE
[init refactor] organize tests

### PR DESCRIPTION
**Pull request by balopat**
This should come after #3538 is merged, it's built on top of that. 

- It reorganizes the tests from init_test.go to the relevant file's tests.
- adds tests for `stripTags` that is a new method introduced as I'm starting to take `DoInit` apart

I know...this is a bit bigger but I am close to getting to a sensible organization of the code. 
Also, after merging #3538 this should become smaller.

### CODE REVIEW

**Comment by balopat** [comment]
Thank you for all the reviews @dgageot! You are a rock star! ⭐️

File: pkg/skaffold/initializer/analyze.go [hunk 1]
```diff
@@ -31,6 +31,26 @@ type analysis struct {
...
+func newAnalysis(c Config) *analysis {
+	var builders *builderAnalyzer
+	if !c.SkipBuild {
+		builders = &builderAnalyzer{
+			findBuilders:        !c.SkipBuild,
+			enableJibInit:       c.EnableJibInit,
+			enableBuildpackInit: c.EnableBuildpackInit,
+		}
+	}
+	a := &analysis{
```
Comment (dgageot) [inline pkg/skaffold/initializer/analyze.go]: can we inline that?

File: pkg/skaffold/initializer/analyze.go [hunk 2]
```diff
@@ -105,14 +125,14 @@ type directoryAnalyzer struct {
 	currentDir string
 }
 
-func (a *directoryAnalyzer) analyzeFile(filePath string) error {
+func (a *directoryAnalyzer) analyzeFile(_ string) error {
```
Comment (dgageot) [inline pkg/skaffold/initializer/analyze.go]: `_` can be removed, right?

File: pkg/skaffold/initializer/builders.go [hunk 3]
```diff
@@ -214,6 +220,14 @@ func resolveBuilderImages(builderConfigs []InitBuilder, images []string, force b
 		return nil, errors.New("unable to automatically resolve builder/image pairs; run `skaffold init` without `--force` to manually resolve ambiguities")
 	}
 
+	pairs, err := resolveBuilderImagesInteractively(builderConfigs, images)
```
Comment (dgageot) [inline pkg/skaffold/initializer/builders.go]: `return resolveBuilderImagesInteractively(builderConfigs, images)`?

File: pkg/skaffold/initializer/analyze.go (hunk 2 above)
Comment (balopat) [reply pkg/skaffold/initializer/analyze.go]: it's not going to implement Analyzer then :/

Comment (dgageot) [reply pkg/skaffold/initializer/analyze.go]: I mean keep the argument but don’t give it a _ name.

**Pull request merged**